src/xbot/common/math/PID.java

```diff
@@ -102,17 +142,24 @@ public double calculate(double goal, double current,
             double p, double i, double d) {
         return calculate(goal, current, p, i, d, 0);
     }
-
+    
     /**
-     * This tells you if the controller has met its goal. Don't call this before
-     * calling calculate!
-     * 
-     * @param absoluteTolerance
-     *            How close the value can be before it is considered
-     *            "on-target."
+     * Check for all conditions where the tolerance is above 0. Since tolerances default to
+     * -1, it will skip any unassigned ones.
      */
-    public boolean isOnTarget(double absoluteTolerance)
-    {
-        return Math.abs(m_targetInputValue - m_currentInputValue) < absoluteTolerance;
+    public boolean isOnTarget() {
+        boolean checkError = errorTolerance > 0;
+        boolean checkDerivative = derivativeTolerance > 0;
+        
+        boolean isOnTarget = true;
```

Comment:
I can imagine a functional case where we'd want a command using this logic to run indefinitely: you have a button that rotates the robot to a certain angle and stays there, On the other hand, having it end _immediately_ doesn't have a use case -- it doesn't do anything. To me, it seems like it's a lot easier to figure out the answer to "why is my robot stuck trying to point in this direction" than "why did my robot not point where it was supposed to?". If it never stops, at least you can see it live on the dashboard; if it exits immediately, you have to find the info in logs and realize that something finished too early.